docs/explanations/release/release.md

```diff
@@ -0,0 +1,54 @@
+Everyone should be able to do a release. Ideally most of the steps should be automated. We have prepared an ISSUE_TEMPLATE with a "Release Checklist" that should be easy to follow, highlighting all the steps and useful links.
+
+The Product Release should contain important notes, the changelog with the list of new features, bug fixes and the corresponding links to PRs and issues.
+It should also contains a reference to the Helm Charts, images and binaries previously released and built.
+
+The publishing of the Product Release Notes is a manual step.  
+
+
+## Current state
+
+Epinio is made up of different components.
+
+The [`epinio/epinio`](https://github.com/epinio/epinio) repository contains the code used to build the binary of the `epinio` CLI and the `epinio-server` image.
+
+The Epinio UI image is made from the [`ui-backend`](https://github.com/epinio/ui-backend) (a proxy for the [`dashboard`](https://github.com/epinio/ui). The dashboard is hosted in the `epinio-dev` branch of the [`rancher/dashboard`](https://github.com/rancher/dashboard) repository.
+
+The [`helm-charts`](https://github.com/epinio/helm-charts) are the suggested and supported way to deploy Epinio. They are wrapping the [`epinio`](https://artifacthub.io/packages/helm/epinio/epinio) charts, the [`epinio-ui`](https://artifacthub.io/packages/helm/epinio/epinio-ui) charts, and the dependencies.
+
+
+### Release flow
+
+The following diagram is showing the current release flow. The entrypoint is the Product Release. The publishing of the release will tag `epinio` repository triggering a build. This build will prepare the binaries and images. It will also notify external repositories about this new version. Please note that the Epinio Helm Charts are not released yet, and so even if we published a new release it won't be available until we release them.  
+
+Also the Epinio UI is a chart dependency of the Epinio Helm Chart, and it's following an [independent release](https://github.com/epinio/ui/tree/dev/docs/developer#standalone-1). But since the Product Release is tight to a Epinio release in order to update the UI we still need to tag `epinio`.
+
+![current-release](current-release.png)
+
+
+## Proposal
+
+The version of the Product Release could diverge from the `epinio` or `epinio-ui` versions. It could be aligned with the Helm Charts version because they are the way on how we are supporting the Epinio deployment.
+
+Independent releases and tags of the different components should be possible.
+
+This diagram tries to show a cleaner release flow.
+
+![future-release](future-release.png)
+
+The `epinio` and `epinio-ui` components are built from tags, and the two should be completely independent and automated. The new available images should trigger a PR on the Helm Charts. When ready we could manually trigger an Helm Chart release to publish the new charts with the new images. This release could prepare the draft for the Product Release.
```

Comment:
```suggestion
The `epinio` and `epinio-ui` components are built from tags, and the two should be completely independent and automated. The new available images should trigger a PR on the Helm Charts. When ready we could manually trigger a Helm Chart release to publish the new charts with the new images. This release could prepare the draft for the Product Release.
```

My understanding from the above is that the "Product releases" coincide with the "Helm Chart Releases" we have on github, in the `helm-charts` repository. Is that correct ? I would be in agreement of using that as our product release point. The main thing to spruce up there are the missing references to the component releases, i.e. `epinio`, `epinio-ui`, and dependencies.

The docs would likely have to be modified as well to point to this. (Are they even pointing somewhere for "Product Releases" ?)

Reply:
Another point of interest, I believe, would be the role of the `updatecli` tool in the automation, i.e. inter-repository triggers and the like. It already is doing something in that regard IIRC/IIUC.

Reply:
That's correct, "Product Releseas" are "Helm Chart Releases", because that's the way on how we are telling the users to install Epinio, and saying that the `epinio-server` with x.y.z works well with ui a.b.c and these dependencies.

Since we are using the same version at the moment the docs are already pointing to the Helm Chart release, but yes, in the future they could point to a different version, and maybe it will make sense to have an entrypoint that shows in a clear way that the Epinio release 2.3.5 is made of `epinio-server 2.1.4` and `epinio-ui 2.3.3` for example.

Would you change this part in a different way to make it clearer, or is it okish?

About `updatecli` I didn't want to mention any specific tool, but yes, it's basically the "check dependency update" arrow.

Reply:
About "This release consists of these pieces ...", I believe that it would be sufficient to simply add such information to a the helm chart github release, like https://github.com/epinio/helm-charts/releases/tag/epinio-1.5.1

The section is okish.